src/Layouts/Layout.tsx

```diff
@@ -13,29 +14,46 @@ import logout_nav_icon from "../assets/Images/Logout_nav_icon.png";
 import bell_icon from "../assets/Images/Bell_icon.png";
 import shop_icon from "../assets/Images/shop_icon.png";
 import NavItems from "../Components/Elements/NavItems";
-import { useState } from "react";
+import { useContext, useEffect, useState } from "react";
 import Counter from "../Components/Elements/Counter";
-import profile_pic from "../assets/Images/Profile_pic.png";
 import SearchBox from "../Components/Elements/SearchBox";
 import Profile from "../Components/Elements/Profile";
 import menu_icon from "../assets/Images/men_icon.png";
 import close_icon from "../assets/Images/close_icon.png";
+import UserContext from "../contexts/UserContext";
+import { baseURL } from "../const/const";
+import axios from "axios";
 
 const Layout = () => {
   const [notifications, setNotifications] = useState(5);
   const [shopCount, setShopCount] = useState(8);
   const [search, setSearch] = useState("");
-  const [user, setUser] = useState({ name: "Test User", image: profile_pic });
   const [active, setActive] = useState("active");
   const [mobileMenuActive, setMobileMenuActive] = useState(false);
+  const { user, setUser } = useContext(UserContext);
+  const navigate = useNavigate();
+  const logout = async () => {
+    const result = await axios.get(baseURL + "/api/v1/auth/logout", {
+      withCredentials: true,
+    });
+    if (result.status === 200) {
+      setUser(result.data);
+      navigate("/login");
+    }
+  };
+  useEffect(() => {
+    if (!user.first_name) {
+      navigate("/login");
+    }
+  });
```

Comment:
useEffect logic needs to be updated here,

Reply:
noted. I'll change it